src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -46,16 +48,23 @@ public function __construct(ControllerNameParser $parser, LoggerInterface $logge
     }
 
     /**
-     * Loads a resource.
-     *
-     * @param mixed  $resource A resource
-     * @param string $type     The resource type
-     *
-     * @return RouteCollection A RouteCollection instance
+     * {@inheritdoc}
      */
     public function load($resource, $type = null)
     {
-        $collection = parent::load($resource, $type);
+        if (false !== $this->loading) {
+            // This can happen if a fatal error occurs in parent::load()
```

Comment:
This is specific to the FrameworkBundle because fatal error re-injection is configured in the service definition. If someone has a different loader, the failure will work but the degraded exception page, like today.